python/lsst/afw/coord/refraction.py

```diff
@@ -0,0 +1,284 @@
+    elevation : lsst.afw.geom Angle
+        Elevation of the observation, as an Angle.
+    weather : lsst.afw.coord Weather, optional
+        Class containing the measured temperature, pressure, and humidity
+        at the observatory during an observation
+    observatory : lsst.afw.coord Observatory, optional
+        Class containing the longitude, latitude, and altitude of the observatory.
+
+    Returns
+    -------
+    lsst.afw.geom Angle
+        The refraction at `wavelength` - the refraction at `wavelengthRef`.
+    """
+    refractionStart = refraction(wavelength, elevation, weather=weather, observatory=observatory)
+    refractionEnd = refraction(wavelengthRef, elevation, weather=weather, observatory=observatory)
+    return refractionStart - refractionEnd
+
+
+def deltaN(wavelength, weather):
+    """Calculate the differential refractive index of air.
+
+    The differential refractive index is the difference of the refractive index from 1.,
+    multiplied by 1E8 to simplify the notation and equations.
+    Calculated as (n_air - 1)*10^8
+
+    This replicates equation 14 of Stone 1996 "An Accurate Method for Computing Atmospheric Refraction"
+
+    Parameters
+    ----------
+    wavelength : float
+        wavelength is in nanometers
+    weather : lsst.afw.coord Weather
+        Class containing the measured temperature, pressure, and humidity
+        at the observatory during an observation
+
+    Returns
+    -------
+    float
+        The difference of the refractive index of air from 1., calculated as (n_air - 1)*10^8
+    """
+    waveNum = 1E3/wavelength  # want wave number in units 1/micron
+
+    dryAirTerm = 2371.34 + (683939.7/(130. - waveNum**2.)) + (4547.3/(38.9 - waveNum**2.))
+
+    wetAirTerm = 6487.31 + 58.058*waveNum**2. - 0.71150*waveNum**4. + 0.08851*waveNum**6.
+
+    return (dryAirTerm*densityFactorDry(weather) +
+            wetAirTerm*densityFactorWater(weather))
+
+
+def densityFactorDry(weather):
+    """Calculate dry air pressure term to refractive index calculation.
+
+    This replicates equation 15 of Stone 1996 "An Accurate Method for Computing Atmospheric Refraction"
+
+    Parameters
+    ----------
+    weather : lsst.afw.coord Weather
+        Class containing the measured temperature, pressure, and humidity
+        at the observatory during an observation
+
+    Returns
+    -------
+    float
+        Returns the relative density of dry air at the given pressure and temperature.
+    """
+    temperature = _extractTemperature(weather, useKelvin=True)
+    waterVaporPressure = humidityToPressure(weather)
+    airPressure = _extractPressure(weather)
+    dryPressure = airPressure - waterVaporPressure
+
+    eqn = (dryPressure/cds.mbar)*(57.90E-8 - 9.3250E-4*u.Kelvin/temperature +
+                                  0.25844*u.Kelvin**2/temperature**2.)
+
+    densityFactor = float((1. + eqn)*(dryPressure/cds.mbar)/(temperature/u.Kelvin))
+
+    return densityFactor
```

Comment:
There's nothing terribly wrong with it, but why do you like to do:

    value = calculate(something)
    
    return value

instead of:

    return calculate(something)

?

Reply:
I find it helpful for debugging since I can easily add a breakpoint or print statement, and that it makes the code more readable. But it's mostly a habit, and one I'm happy to change when there are performance or memory issues.

Reply:
Fair enough. No performance or memory issues I'm aware of so do whatever you're comfortable with.